tensorflow/compiler/xla/service/gpu/matmul_utils.cc

```diff
@@ -297,6 +297,8 @@ StatusOr<bool> CanFoldTransposeOperandIntoDot(const HloInstruction& dot,
     TF_RET_CHECK(lhs_layout.num_cols == rhs_layout.num_rows);
     TF_RET_CHECK(output_layout.num_rows == lhs_layout.num_rows);
     TF_RET_CHECK(output_layout.num_cols == rhs_layout.num_cols);
+    TF_RET_CHECK(c_layout.num_rows == output_layout.num_rows);
+    TF_RET_CHECK(c_layout.num_cols == output_layout.num_cols);
   } else {
     TF_RET_CHECK(lhs_layout.num_cols == rhs_layout.num_cols);
     TF_RET_CHECK(output_layout.num_rows == lhs_layout.num_rows);
```

Comment:
Also check `c_layout` in this else branch.

Reply:
I think we should have a flag. Right now it's messy how GemmConfig either the LHS or the RHS to be transposed depending on if the output is transposed, and depending on whether the GemmConfig is passed to `MatmulPlan` or just  used to query the output layout.

But we can deal with this in a later change. Since we want to get this change in ASAP and this is my only issue with the PR, I'll approve and merge it.